### ISSUE TITLE
SMART on FHIR authorization support not reported in the conformance document

### ISSUE THREAD

**Issue by emcrisostomo** [smart-on-fhir/smart-dev-sandbox#3]
Hi,

I expected the sandbox to list the OAuth2 token and authorization endpoints in the conformance document (http://docs.smarthealthit.org/authorization/conformance-statement/): I've checked both the hosted sandbox and these containers but they do not report it.

If the sandbox is meant to 

> The SMART Health IT Sandbox is a virtual testing environment that mimics a live EHR production environment [...]

shouldn't it then support that feature? Or am I missing something?

Thanks.

**Comment by gotdan** [comment]
The open FHIR endpoint of the sandbox doesn't display this information. However, if you get the conformance statement from a secure endpoint (either the iss parameter passed as part of an EHR launch or the url displayed in the bottom section of the UI for a standalone launch) it should be there. I'll close this issue for now, but feel free to reopen it if you're still not seeing the auth endpoints.

**Comment by emcrisostomo** [comment]
Thank you @gotdan. I still have a couple of questions though. I'm using the sandbox because I'm programmatically invoking the SMART on FHIR authentication flow. In particular: in a standalone launch sequence where I'm supposed to GET `/metadata` directly, without getting the `iss` as part of the EHR launch sequence, how am I supposed to discover then the full conformance statement path? I don't find it documented in this sandbox either.

Thanks.

**Comment by vlad-ignatov** [comment]
The conformance statement is always at `/metadata` so for standalone launches you can just get the value of the `FHIR Server Url` field at the bottom and append `/metadata` to it. The result would be something like http://launch.smarthealthit.org/v/r2/sim/eyJoIjoiMSIsImoiOiIxIn0/fhir/metadata

**Comment by vlad-ignatov** [comment]
Also, the difference between EHR and standalone launches is that in EHR mode the app is notified where the server is at launch time (via the `iss` parameter), while the standalone apps are "pre-configured" to know their server's location. In other words, get that `FHIR Server Url` and use it as you wish in your code - hardcode it somewhere, add it to your configuration, etc.

For web apps you can use the fhir-client JS library. Here is an example of how to do a standalone app launcher:  https://github.com/smart-on-fhir/sample-apps-stu3/blob/master/medications-standalone/launch.html

**Comment by emcrisostomo** [comment]
Thanks @vlad-ignatov, that URL in effect returns the OAuth2 token and authorise endpoint. I still don't understand how to derive it though. For the sake of argument, let's consider the local sandbox. When I navigate to http://localhost:4000/ one of the URLs I see for the FHIR endpoints is http://localhost:4002/baseDstu2/metadata. I assumed that was the `/metadata` I would hit, since the console also says that

```
Server Base URL: http://localhost:4002/baseDstu2/
```

is the base URL (hence my first message when I opened this issue).

Basically, I can't find the `FHIR Server Url` you mention.

Thanks again.

**Comment by vlad-ignatov** [comment]
I see. Here is how that works (I will use real links but the local sandbox will follow the same scheme):

1. We have these two fhir servers for DSU2 and STU3 located at https://r2.smarthealthit.org and https://r3.smarthealthit.org. They are both open, thus you will not find those authorization endpoints in their conformance statement. 
2. Then we have the launcher sitting in front of these two servers, acting as a proxy - https://launch.smarthealthit.org/. If you open it and select one of the two standalone launch types, you will see the generated `FHIR Server Url` at the bottom. For example https://launch.smarthealthit.org/?auth_error=&fhir_version_1=r2&fhir_version_2=r2&iss=&launch_prov=1&launch_url=&patient=&prov_skip_auth=1&provider=&pt_skip_auth=1&public_key=&sb=&sde=&token_lifetime=15&user_pt=

The example I gave you will use https://r2.smarthealthit.org internally as FHIR server. It will add the SMART authorization layer (on top of that otherwise open fhir server) and will modify the conformance statement so that apps can discover auth endpoints...

Finally, here a screenshot of the field I was talking about:
<img width="760" alt="screen shot 2018-06-28 at 4 37 53 pm" src="https://user-images.githubusercontent.com/1119082/42059536-c85e6234-7af1-11e8-894e-a3c3ab9ae523.png">
